src/main/java/com/bootcamp/demo/service/userDetails/UserDetailsImpl.java

```diff
@@ -0,0 +1,92 @@
+package com.bootcamp.demo.service.userDetails;
+
+import com.bootcamp.demo.model.User;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class UserDetailsImpl implements UserDetails {
+     private final String id;
+     private final String email;
+     @JsonIgnore
+     private final String password;
+
+     private final Collection<? extends GrantedAuthority> authorities ;
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return authorities;
+    }
+
+   public UserDetailsImpl(String id, String email, String password, Collection<? extends GrantedAuthority> authorities){
+        this.id = id;
+        this.email = email;
+       this.password = password;
+       this.authorities = authorities;
+   }
```

Comment:
This code is not properly formatted.

Reply:
Resolved here : 1badabc5cf63827dbdacc8858d74670f7a3bb5b7